### PR TITLE
add qiime qzv visualisation extractor

### PR DESCRIPTION
**Pull request by bernt-matthias**
the qiime2 display application (view.qiime2.org) does not work for non-public Galaxy instances. This is a workaround.

qiime2 viz artifacts are just zip files which also contain the html + data + files (which is shown at the view website). So this tool is equivalent

If I find the time I will explore the possibility to do the same with a generic / viz plugin .. at least for so long (and maybe for workflows) the tool will be useful.

Thanks to @EngyNasr for the discussion and help.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/qiime_extract_viz/qiime_extract_viz.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        for nonjs in \$(cat '${viz.extra_files_path}'/*.html | grep '<script src="' | sed 's/^\s*<script src="//; s/".*//' | grep -v "\.js\$"); do
+            js=\$(echo "\$nonjs" | sed 's/\.[^.]\+$//').js &&
+            mv '${viz.extra_files_path}'"/\$nonjs" '${viz.extra_files_path}'"/\$js" &&
+            sed -i -e "s@\"\${nonjs}\"@\"\${js}\"@" '${viz.extra_files_path}'/*.html;
+        done &&
+
+        mv '${viz.extra_files_path}'/index.html '$viz'
+    ]]></command>
+    <inputs>
+        <param name="infile" type="data" format="qzv" label="QIIME visualization artifact" />
```
Comment (bernt-matthias) [inline tools/qiime_extract_viz/qiime_extract_viz.xml:34]: Qza?

Comment (bernt-matthias) [reply tools/qiime_extract_viz/qiime_extract_viz.xml:34]: Seem to contain only data. Metadata should be accessible also without the tool. So qzv should be sufficient.

File: tools/qiime_extract_viz/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: qiime_extract_viz
+owner: iuc
+description: Extract vizualization from QIIME artifacts
+long_description: |
+  TODO
```
Comment (pvanheus) [inline tools/qiime_extract_viz/.shed.yml:5]: This needs to be completed before merge.

File: tools/qiime_extract_viz/qiime_extract_viz.xml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+
+.. class:: infomark
+
+**What it does**
+
+The tool extracts visualisation from QIIME artivarct datasets (qza, qzv).
```
Comment (pvanheus) [inline tools/qiime_extract_viz/qiime_extract_viz.xml:65]: Do you mean "QIIME artifacts"?

Comment (bernt-matthias) [reply tools/qiime_extract_viz/qiime_extract_viz.xml:65]: Indeed. My bad 😅

File: tools/qiime_extract_viz/.shed.yml (hunk 2 above)
Comment (bernt-matthias) [reply tools/qiime_extract_viz/.shed.yml:5]: Thanks for checking carefully. I would have forgotten about this.

**Comment by bernt-matthias** [comment]
Put this on draft .. want to finish the discussion with the qiime2 developers first https://github.com/qiime2/galaxy-tools/issues/38#issuecomment-1272332723

**Comment by shiltemann** [comment]
@bernt-matthias I just wrote this exact same tool, then found this PR. What are your thoughts on this currently? Should we try to get this in? 

The "display at qiime2view" links are still broken (https://github.com/galaxyproject/galaxy/issues/17658) in Galaxy, pasting shared dataset URLs in qiime2view manually is also broken due to CORS settings, so currently no good way to view qzv files in/from Galaxy. Would be nice if the qiime wrappers output html outputs such as these by default, but until then this at least provides a way for users now to view qiime qzv outputs?

**Comment by bernt-matthias** [comment]
> I just wrote this exact same tool, then found this PR. What are your thoughts on this currently? Should we try to get this in?

Sorry for that. We should have merged this a long time ago. Lets go ahead. 

Any comments from your side?

**Review by shiltemann — APPROVED** [review]

**Comment by shiltemann** [comment]
@bernt-matthias ok great, just tested your tool locally and looks good to me :+1:

**Pull request merged**